django/contrib/auth/middleware.py

```diff
@@ -12,6 +16,12 @@ def get_user(request):
     return request._cached_user
 
 
+async def auser(request):
+    if not hasattr(request, "_cached_user"):
+        request._cached_user = await sync_to_async(auth.get_user)(request)
+    return request._cached_user
```

Comment:
I'm not sure about sharing the same cache, it looks like a receipt for headache :face_with_head_bandage:  As far as I'm aware, it can only help avoiding an extra query when someone use both `auser` and `user` in the same request-response cycle which should be rather niche :thinking: